go/worker/compute/executor/committee/hooks.go

```diff
@@ -35,12 +33,13 @@ func (n *Node) HandleNewBlockEarlyLocked(*runtime.BlockInfo) {
 // HandleNewBlockLocked implements NodeHooks.
 // Guarded by n.commonNode.CrossNode.
 func (n *Node) HandleNewBlockLocked(bi *runtime.BlockInfo) {
-	// Drop blocks if the worker falls behind.
+	// Stop sending blocks if the worker has stopped.
 	select {
-	case <-n.blockInfoCh:
+	case <-n.stopCh:
+		return
 	default:
 	}
 
-	// Non-blocking send.
+	// Don't skip any blocks, as the round worker must finalize the final round.
```

Comment:
Could instead use a ring channel with a size of 1, so that we always only queue the last block in case the worker falls behind? Or would that still cause issues?

Reply:
Removed this commit, as we don't need to process all blocks. If we skip some, the only problem will be that processed transactions won't be removed from the pool. However, they will be removed latter as their check will fail.